utils/search/search_general.py

```diff
@@ -54,27 +65,36 @@ def search_prepare_sort(sort, options):
 
 
 def search_process_filter(filter_query):
-    # Process the filter to replace humnan-readable Audio Commons descriptor names for the dynamic field names used in
-    # Solr (e.g. ac_tonality -> ac_tonality_s, ac_tempo -> ac_tempo_i). The dynamic field names we define in Solr
-    # schema are '*_b' (for bool), '*_d' (for float), '*_i' (for integer) and '*_s' (for string). At indexing time
-    # we append these suffixes to the ac descirptor names so Solr can treat the types properly. Now we automatically
-    # append the suffices to the filter names so users do not need to deal with that.
+    """Process the filter to replace humnan-readable Audio Commons descriptor names.
+
+    Used for the dynamic field names used in Solr (e.g. ac_tonality -> ac_tonality_s, ac_tempo -> ac_tempo_i).
+    The dynamic field names we define in Solr schema are '*_b' (for bool), '*_d' (for float), '*_i' (for integer) 
+    and '*_s' (for string). At indexing time we append these suffixes to the ac descirptor names so Solr can 
+    treat the types properly. Now we automatically append the suffices to the filter names so users do not 
+    need to deal with that.
+
+    Args:
+        filter_query (str): query filter string.
+
+    Returns:
+        str: processed filter query string.
+    """
     for name, t in settings.AUDIOCOMMONS_INCLUDED_DESCRIPTOR_NAMES_TYPES:
         filter_query = filter_query.replace('ac_{0}:'.format(name), 'ac_{0}{1}:'
                                             .format(name, settings.SOLR_DYNAMIC_FIELDS_SUFFIX_MAP[t]))
     return filter_query
 
 
-def search_prepare_parameters(request):
-    """ Process the query parameters.
-        Returns: - the params for search_prepare_query() in a dict
-                 - the variables for the template context
-    """
-    search_query = request.GET.get("q", "")
-    filter_query = request.GET.get("f", "")
-    filter_query_link_more_when_grouping_packs = filter_query.replace(' ','+')
-    cluster_id = request.GET.get('cluster_id', "")
+def split_filter_query(filter_query, cluster_id):
+    """Splits query filters.
```

Comment:
This docstring should be a bit more detailed. The really useful information is in the return line. I guess this function is useful for the template to display the filter and the link to remove it right? also the fact that `cluster_id` is treated separately should be explained in the docstring as well instead of only in code comment as it is quite relevant.